rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py

```diff
@@ -550,6 +549,14 @@ def filter_waypoints(self, wps: list, threshold=1.0):
                 waypoints[index].position = last_pose
                 index = index + 1
 
+        # If robot is already past the first waypoint, remove first waypoint
+        if len(waypoints) > 1:
+            dist_to_next = self.dist(waypoints[1].position, last_pose)
+            first_to_next = \
+                self.dist(waypoints[0].position, waypoints[1].position)
+            if dist_to_next < first_to_next:
+                del waypoints[0]
```

Comment:
We do not want to delete any waypoints, especially the first one as the fleet adapter could have been interrupted with a new path where the robot is expected to wait exactly where it was to avoid conflicts with other robots. So the expectation here is that the robot will need to wait at that first waypoint until a certain time has passed before proceeding to the next. But with this implementation, the robot will go to the second waypoint without waiting.